c/parallel/test/test_merge_sort.cpp

```diff
@@ -432,12 +432,16 @@ C2H_TEST("DeviceMergeSort::SortKeys works with output iterators", "[merge_sort]"
     make_iterator<TestType, random_access_iterator_state_t>(
       {"random_access_iterator_state_t", "struct random_access_iterator_state_t { int* d_input; };\n"},
       {"advance",
-       "extern \"C\" __device__ void advance(random_access_iterator_state_t* state, unsigned long long offset) {\n"
-       "  state->d_input += offset;\n"
+       "extern \"C\" __device__ void advance(void* state, const void* offset) {\n"
+       "  auto* typed_state = static_cast<random_access_iterator_state_t*>(state);\n"
+       "  auto offset_val = *static_cast<const unsigned long long*>(offset);\n"
+       "  typed_state->d_input += offset_val;\n"
```

Comment:
Question: why not use a raw string literal?

Reply:
We can. Changed to using raw strings in https://github.com/NVIDIA/cccl/pull/6634/commits/ac415014c58353f012ab5776b7f8ec3588f35f12.